Suggest to "Open the singleton class with `class << self`" when `Style/ModuleFunction` is set to `forbidden`

RDeckard opened the pull request:
**Context:**
``` yml
Style/ModuleFunction:
  EnforcedStyle: forbidden
```

When `extend self` or `module_function` is used in this context ⬆️, the cop message might suggest to "_Open the singleton class with `class << self` instead._".
WDYT?

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

RDeckard commented:
[The broken `ruby-2.6-spec` CI step](https://circleci.com/gh/rubocop/rubocop/242160) might be flacky?

bbatsov commented:
We rarely use messages that are multi-sentence, so I think it would probably be better to improve the cop documentation instead. I see we have no suggestions for the forbidden style at all there and in general the documentation is pretty basic to begin with.

RDeckard commented:
> I think it would probably be better to improve the cop documentation instead.

Good idea, I've reverted the changes and just updated the documentation. 👍

bbatsov commented:
Thanks!
